src/processor/operator/persistent/reader/file_error_handler.cpp

```diff
@@ -39,6 +39,9 @@ void SharedFileErrorHandler::tryCacheError(CopyFromFileError error, const common
 }
 
 void SharedFileErrorHandler::handleError(CopyFromFileError error) {
+    if (error.mustThrow) {
```

Comment:
I would have @royi-luo take a look here. I thought we should put this after lock?